### DISCUSSION TITLE
AttributeError at / 'QuerySet' object has no attribute 'likes' in Django

### DISCUSSION THREAD

**Discussion by LuigiP99** [community/community#37392] [Other Feature Feedback, Questions, & Ideas] [answered]
Hello people, to start I must say that I am new to Django and I am practicing, so I may be having more errors, if so let me know, I would really appreciate it.

I have these models in my app, the image template contains an image and the likes it may have, and the user likes template contains the likes of a user

```
from django.db import models
from django.core.validators import FileExtensionValidator
from django.contrib.auth.models import User

# Create your models here.

class UserLike(models.Model):
    user_like = models.ForeignKey(User, on_delete=models.SET_NULL, null=True)

class Images(models.Model):
    image = models.ImageField(validators=[FileExtensionValidator])
    likes = models.ManyToManyField(UserLike, blank=True)
    
    def __str__(self):
        return f'Image: {self.image}'
```

I want to add a like in this way:

```
def index(request):
    user = User.objects.get(username=request.user)
    user_like = UserLike.objects.create(user_like=user)
    images = Images.objects.filter(id='6')
    image_like = images.likes.add(user_like)
    image_like.save()
    return render(request, 'index.html', {
        'images': images,
        })
```

But throws me the error:

`AttributeError at /
'QuerySet' object has no attribute 'likes'`

On line:

`image_like = images.likes.add(user_like)`

And i can't understand the class object 'images' don't let me access to the 'likes' attribute

**Comment by Ivanrputra** [comment, marked as answer]
"images = Images.objects.filter(id='6')"
Hi since his line return a queryset but not one record then you get that error. its better to use "get" method like this instead to get one query and not queryset.
"images = Images.objects.get(id='6')"

Hope this help

**Reply by Ttech24-dev** [reply]
I had a similar problem and am really confused.
class Flight(models.Model):
    origin = models.ForeignKey(Airport, on_delete=models.CASCADE, related_name="departures")
    destination = models.ForeignKey(Airport, on_delete=models.CASCADE, related_name="arrivals")
    duration = models.IntegerField()
    
    def __str__(self):
        return f"{self.origin} to {self.destination}" 

class Passenger(models.Model):
    first_name = models.CharField(max_length=64, null=True)
    last_name = models.CharField(max_length=64, null=True)
    phone = models.CharField(max_length=60, null=True)
    email = models.CharField(max_length=60, null=True)
    address = models.CharField(max_length=300, null=True)
    next_of_kin = models.CharField(max_length=60, null=True)
    kin_phone = models.CharField(max_length=60, null=True)
    flights = models.ForeignKey(Flight, on_delete=models.CASCADE, related_name="passengers", null=True)
    
    def __str__(self) -> str:
        return f"{self.last_name} {self.first_name}"

And I tried this query "flight = Flight.objects.all() ,
passenger1= flight.passenger_set.all() "
but am getting an error"QuerySet' object has no attribute 'passenger_set'"
So can you help me???

**Comment by iamadityaupadhyay** [comment]
the best way to solve the problem def update(request,pk):
    queryset=Recipe.objects.filtter(id=pk)
    if request.method=="POST":
        data = request.POST
        queryset.recipe_name=data.get("recipe_name")
        queryset.recipe_description=data.get('recipe_description')
        queryset.recipe_image=data.get("recipe_image")
        queryset.save()
        return redirect("/blog/view-recipe/")
    return render(request,"crud/update_recipe.html",{'recipes':queryset})
I was having the same error but when I got the error but by just replacing the fucky "filter"->"get" it solved .....uh its took me 1 hr
